great_expectations/rule_based_profiler/parameter_builder/value_set_multi_batch_parameter_builder.py

```diff
@@ -0,0 +1,130 @@
+        batch_list: Optional[List[Batch]] = None,
+        batch_request: Optional[Union[BatchRequest, RuntimeBatchRequest, dict]] = None,
+    ):
+        """
+
+        Args:
+            name: the name of this parameter -- this is user-specified parameter name (from configuration);
+            it is not the fully-qualified parameter name; a fully-qualified parameter name must start with "$parameter."
+            and may contain one or more subsequent parts (e.g., "$parameter.<my_param_from_config>.<metric_name>").
+            metric_domain_kwargs: used in MetricConfiguration
+            metric_value_kwargs: used in MetricConfiguration
+            data_context: DataContext
+            batch_list: explicitly passed Batch objects for parameter computation (take precedence over batch_request).
+            batch_request: specified in ParameterBuilder configuration to get Batch objects for parameter computation.
+        """
+        super().__init__(
+            name=name,
+            metric_name="column.distinct_values",
+            data_context=data_context,
+            batch_list=batch_list,
+            batch_request=batch_request,
+            reduce_scalar_metric=False,
+            enforce_numeric_metric=False,
+        )
+
+        self._metric_domain_kwargs = metric_domain_kwargs
+        self._metric_value_kwargs = metric_value_kwargs
+
+    """
+    Full getter/setter accessors for needed properties are for configuring MetricMultiBatchParameterBuilder dynamically.
+    """
+
+    @property
+    def metric_domain_kwargs(self) -> Optional[Union[str, dict]]:
+        return self._metric_domain_kwargs
+
+    @property
+    def metric_value_kwargs(self) -> Optional[Union[str, dict]]:
+        return self._metric_value_kwargs
+
+    @metric_value_kwargs.setter
+    def metric_value_kwargs(self, value: Optional[Union[str, dict]]) -> None:
+        self._metric_value_kwargs = value
+
+    def _build_parameters(
+        self,
+        parameter_container: ParameterContainer,
+        domain: Domain,
+        variables: Optional[ParameterContainer] = None,
+        parameters: Optional[Dict[str, ParameterContainer]] = None,
+    ) -> ParameterContainer:
+
+        # Build the list of unique values for each batch
+        super()._build_parameters(
+            parameter_container=parameter_container,
+            domain=domain,
+            variables=variables,
+            parameters=parameters,
+        )
+
+        # Retrieve and replace the list of unique values for each batch with
+        # the set of unique values for all batches in the given domain.
+        fully_qualified_parameter_name: str = f"$parameter.{self.name}"
+        parameter_value_node: ParameterNode = (
+            get_parameter_value_by_fully_qualified_parameter_name(
+                fully_qualified_parameter_name=fully_qualified_parameter_name,
+                domain=domain,
+                parameters={domain.id: parameter_container},
```

Comment:
@anthonyburdi Also, can we please try the "whole method" 
```
 build_parameter_container(
            parameter_container=parameter_container, parameter_values=parameter_values
        )
```
using the new values?  It might be a safer way as compared to updating a specific value.  Thanks!

Reply:
`parameters=parameters` does not work because here parameters is None if not supplied. Also I looked at that `build_parameter_container()` method but it seems like it will not overwrite the node but please let me know if I am misunderstanding. Trying it now to double check.

Reply:
I have not been able to get this to work as you've described, perhaps we can discuss.

Reply:
@anthonyburdi Could we please make a (very time-boxed) attempt together?  Thanks!